### ISSUE TITLE
Pdf content disappears on export

### ISSUE THREAD

**Issue by Esokrates** [xournalpp/xournalpp#2998]
(Please complete the following information, and then delete this line)

**Affects versions :**
 - OS: Linux
 - (Linux only) Desktop environment: Gnome-Wayland
 - Which version of libgtk do you use
     3.24.21
 - Version of Xournal++: 
     1.0.20
 - Installation method: Flatpak

**Describe the bug**
Importing and exporting the attached pdf file does not work as expected: Most of the pdf content simply disappears in the export. This makes filling out the pdf impossible with xournal++.

[Einverständniserklärung.pdf](https://github.com/xournalpp/xournalpp/files/6252465/Einverstandniserklarung.pdf)

**To Reproduce**
1. Open attached pdf
2. Export as pdf

**Expected behavior**
The exported pdf should look like the imported one.

**Screenshots of Problem**
Original pdf:
![Screenshot from 2021-04-03 13-27-36](https://user-images.githubusercontent.com/17780185/113477168-64097c00-9480-11eb-84d9-f568f2f1c553.png)
Exported pdf:
![Screenshot from 2021-04-03 13-27-20](https://user-images.githubusercontent.com/17780185/113477170-67046c80-9480-11eb-8def-24ce824ef67a.png)

**Comment by Febbe** [comment]
Thank you for the attachment!

**Comment by Febbe** [comment]
Also happens on newest master.
```text
febbe:~/Downloads$ pdffonts Einverstandniserklarung.pdf 
name                                 type              encoding         emb sub uni object ID
------------------------------------ ----------------- ---------------- --- --- --- ---------
SYOGRD+StRyde-Medium                 Type 1C           MacRoman         yes yes no       8  0
NYPDQR+HelveticaNeue-Bold            TrueType          MacRoman         yes yes no      10  0
PTUPXF+HelveticaNeue                 TrueType          MacRoman         yes yes no       9  0
VRJWSD+Wingdings2                    TrueType          WinAnsi          yes yes yes     12  0
febbe:~/Downloads$ pdffonts Einverstandniserklarung_export.pdf 
Syntax Error (185): Illegal character '}'
Internal Error: xref num 6 not found but needed, try to reconstruct<0a>
Syntax Error (185): Illegal character '}'
Syntax Error: font resource is not a dictionary
Syntax Error (264): Illegal character '>'
Syntax Error: font resource is not a dictionary
Syntax Error (792): Illegal character '>'
Syntax Error: font resource is not a dictionary
Syntax Error (1806): Illegal character '}'
Syntax Error: font resource is not a dictionary
name                                 type              encoding         emb sub uni object ID
------------------------------------ ----------------- ---------------- --- --- --- ---------
```

**Comment by rolandlo** [comment]
This is most likely a Cairo bug. On Windows it exports fine with the latest master (which uses Cairo 1.17.4-2.). Compare with #2640

**Comment by Febbe** [comment]
It can be, that on windows the fonts are installed. Can you send me the on windows generated document?

**Comment by rolandlo** [comment]
Here it is:
[Einverstandniserklarung_export.pdf](https://github.com/xournalpp/xournalpp/files/6254983/Einverstandniserklarung_export.pdf)

```
roland@roland-ThinkPad-X1-Yoga-3rd:/media/roland/usbdata$ pdffonts Einverstandniserklarung_export.pdf 
name                                 type              encoding         emb sub uni object ID
------------------------------------ ----------------- ---------------- --- --- --- ---------
UCMKVU+StRyde-Medium                 Type 1C           WinAnsi          yes yes yes      6  0
YWOYMN+HelveticaNeue                 TrueType          WinAnsi          yes yes yes      7  0
IQNNIP+HelveticaNeue-Bold            TrueType          WinAnsi          yes yes yes      8  0
XRMCZC+Wingdings2                    TrueType          WinAnsi          yes yes yes      9  0
```

**Comment by Febbe** [comment]
Yes works like a charm. Then this is an issue with a dependency :/. 
Can you find out, if your version of cairo, poppler and gtk is newer?

gtk: 3.24.20
cairo: 1.16.0
poppler: 0.86.1

**Comment by rolandlo** [comment]
@Febbe Not sure, you asked me, but on Windows Azure Pipelines build for the nightly release we use
gtk 3.24.27
cairo 1.17.4-2
poppler: 20.12.1-2 (something with your poppler version number seems wrong)

In general Windows uses more up-to-date versions than Linux (and MacOS). But I think in this case it's all due to Cairo.

**Comment by mjg** [comment]
The exported pdf is fine here (Fedora 33) with xournalpp from git and:
gtk3-3.24.28
cairo-1.16.0
poppler-0.90.0

Poppler's version jumped from 0.90.1 to 20.08.0. Versions 0.90 and below are pretty common.

**Comment by bhennion** [comment]
Is this still happening? It seems like it could have been fixed by #3551 and/or #3939.

**Comment by bhennion** [comment]
@Esokrates Thanks for the feedback. Just to make sure: you are testing the export via the GUI (and not from the terminal directly)?

**Comment by Esokrates** [comment]
@bhennion Yes, export via GUI. FYI: the file to reproduce is attached in the report, so in case I'm not responsive you could test it.
